### ISSUE TITLE
Support for nested structures

### ISSUE THREAD

**Issue by joernlenoch** [caarlos0/env#49]
Hi !

Most recently started using your project for simplifying one of my largest cofiguration structures. Unfortunately there is no support for nested structures, yet.

My first idea would recursively checking the included structs.

```go
type Config struct {
  
  // [...] root config

  Mysql struct {  
     Host string `env:"MYSQL_HOST"`
  }

  Redis struct {  
     Host string `env:"REDIS_HOST"`
  }
}
```

Alternatively, someone could attempt a prefix-esque style to nesting the elements.

```go
type Config struct {
  
  // [...] root config

  Mysql struct {  
     Host string `env:"HOST"` // Should result in "MYSQL_HOST"
  } `env:"MYSQL" `
}
```

I can provide a pull request if you like, but wanted to ask first if something like this would fit into your concept.

Kind regards,
Jörn

**Comment by james-fitzgerald** [comment]
I was hoping for the same thing!

**Comment by dstroot** [comment]
Check out: 
https://github.com/joeshaw/envdecode

**Comment by caarlos0** [comment]
You can open a PR, yes ;)

Thanks

**Comment by andriisoldatenko** [comment]
@joernlenoch Are you still going to work on PR or I can create my PR?

**Comment by joernlenoch** [comment]
Very sorry for the delay, 

Please go ahead - I'm unfortunately swamped right now...

**Comment by netsgnut** [comment]
It seems we now have a PR implementing the first part of @joernlenoch's original proposal (#67), which recursively parses `struct` tags.

**Comment by caarlos0** [comment]
fixed in #69
